collectors/kubernetes/autoconfig/autoconfig.go

```diff
@@ -0,0 +1,166 @@
+package autoconfig
+
+import (
+	"os"
+
+	"emperror.dev/errors"
+
+	"github.com/gezacorp/metadatax/collectors/kubernetes"
+	"github.com/gezacorp/metadatax/collectors/kubernetes/apiserver"
+	"github.com/gezacorp/metadatax/collectors/kubernetes/kubelet"
+)
+
+var AutoConfigurationFailedErr = errors.NewPlain("auto configuration failed")
+
+type SourceType string
+
+const (
+	KubeletSourceType   SourceType = "kubelet"
+	APIServerSourceType SourceType = "apiserver"
+)
+
+type Provider string
+
+const (
+	K3sProvider      Provider = "k3s"
+	KindProvider     Provider = "kind"
+	MicroK8sProvider Provider = "microk8s"
+	GKEProvider      Provider = "gke"
+	EKSProvider      Provider = "eks"
```

Comment:
is there plan to support AKS?

Reply:
sure, i have no AKS cluster to test on unfortunately.